tree/dataframe/src/RDFActionHelpers.cxx

```diff
@@ -163,6 +163,60 @@ template void MeanHelper::Exec(unsigned int, const std::vector<char> &);
 template void MeanHelper::Exec(unsigned int, const std::vector<int> &);
 template void MeanHelper::Exec(unsigned int, const std::vector<unsigned int> &);
 
+StdDevHelper::StdDevHelper(const std::shared_ptr<double> &meanVPtr, const unsigned int nSlots)
+   : fNSlots(nSlots), fResultStdDev(meanVPtr), fDistancesfromMean(nSlots, 0), fMeans(nSlots, 0), fCounts(nSlots, 0)
+{
+}
+
+void StdDevHelper::Exec(unsigned int slot, double v)
+{
+   // Online algorithm
+   auto count = ++fCounts[slot];
+   auto delta = v - fMeans[slot];
+   auto mean = fMeans[slot] + delta / count;
+   auto delta2 = v - mean;
+   auto distance = fDistancesfromMean[slot] + delta * delta2;
+
+   fCounts[slot] = count;
+   fMeans[slot] = mean;
+   fDistancesfromMean[slot] = distance;
+}
```

Comment:
Are we sure of this algorithm? Is it numerically stable? Can we avoid the division? Can we make it incremental?

Reply:
It's the Welford's algorithm. I think it is already incremental, as at the end of each step you know the partial result for the samples up to that point, and by the end you merge everything. I am going to investigate about the division.

>  John D. Cook found the accuracy of this method comparable to the accuracy of the two-pass method derived directly from the definition [cit.](http://jonisalonen.com/2013/deriving-welfords-method-for-computing-variance/)

> This algorithm is much less prone to loss of precision due to catastrophic cancellation, but might not be as efficient because of the division operation inside the loop. [cit.](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm)

Reply:
I followed @stwunsch suggestion and tried to run the Welford's algorithm against the deviation evaluated using its definition. Even for few samples (that should be the most critical part) the difference is in the order of ^-16. I suppose the algorithm is good.

Reply:
@imaxoi that's great. few samples are not necessarily the critical case, many samples might cause problems due to numerical instabilities in the summations.

in any case, since you already have the code, can you make it a gtest?

Reply:
Ok 7e0b87e641eacb69695083fd90c52dd65cdd2b45